Modules/Loadable/Data/qSlicerSceneReader.cxx

```diff
@@ -106,17 +107,42 @@ bool qSlicerSceneReader::load(const qSlicerIO::IOProperties& properties)
     d->CamerasLogic->SetCopyImportedCameras(wasCopying);
     }
 
-  if (this->mrmlScene()->GetLastLoadedVersion() &&
-     this->mrmlScene()->GetVersion() &&
-      strcmp(this->mrmlScene()->GetLastLoadedVersion(),
-             this->mrmlScene()->GetVersion()) > 0 )
+  // Display warning message if scene file was created with a different application or with a future application version
+  std::string currentApplication;
+  int currentMajor = 0;
+  int currentMinor = 0;
+  int currentPatch = 0;
+  int currentRevision = 0;
+  std::string loadedApplication;
+  int loadedMajor = 0;
+  int loadedMinor = 0;
+  int loadedPatch = 0;
+  int loadedRevision = 0;
+  if (vtkMRMLScene::ParseVersion(this->mrmlScene()->GetVersion(), currentApplication,
+    currentMajor, currentMinor, currentPatch, currentRevision)
+    && vtkMRMLScene::ParseVersion(this->mrmlScene()->GetLastLoadedVersion(), loadedApplication,
+      loadedMajor, loadedMinor, loadedPatch, loadedRevision))
     {
-      std::string msg = "Warning: scene file " + file.toStdString() + " has version " +
-        std::string(this->mrmlScene()->GetLastLoadedVersion()) + " greater than Slicer4 version " +
-        std::string(this->mrmlScene()->GetVersion()) + ".";
-
+    QString warningMsg;
+    if (loadedApplication != currentApplication)
+      {
+      warningMsg += tr(" Scene file was saved with %1 application (current application is %2).")
+        .arg(QString::fromUtf8(loadedApplication.c_str()))
+        .arg(QString::fromUtf8(currentApplication.c_str()));
+      }
+    QVersionNumber loadedSceneVersion(loadedMajor, loadedMinor, loadedPatch);
+    QVersionNumber currentVersion(currentMajor, currentMinor, currentPatch);
+    if (loadedSceneVersion > currentVersion)
```

Comment:
Similarly to [QT_VERSION_CHECK](https://github.com/qt/qtbase/blob/e0b89899e3c505edbdece60d6a2d2368a7ef9a01/src/corelib/global/qglobal.h#L61) we could also introduce a `MRML_VERSION_CHECK` macro and then do the following:

```
if (MRML_VERSION_CHECK(loadedMajor, loadedMinor, loadedPatch) > MRML_VERSION_CHECK(currentMajor, currentMinor, currentPatch))
```

Reply:
Similarly, we could have a `SLICER_VERSION_CHECK`.

Reply:
And perhaps in the template an EXTENSION_VERSION_CHECK